src/lib.rs

```diff
@@ -286,6 +293,36 @@ impl Disk {
             Ok(overall)
         }
     }
+
+    // Get the model, firmware, and serial of the disk as a IdentifyParsedDatastruct
+    pub fn identify_parse(&mut self) -> Result<IdentifyParsedData, Errno> {
+        let mut available: SkBool = 0;
+        unsafe {
+            sk_disk_identify_is_available(self.skdisk, &mut available);
+            if available == 1{
+                let parsed_data_pointer: *const SkIdentifyParsedData = null();
+                let ret = sk_disk_identify_parse(self.skdisk, &parsed_data_pointer);
+                if ret < 0 {
+                    let fail = nix::errno::errno();
+                    return Err(Errno::from_i32(fail));
+                }
+                let model = CStr::from_ptr((*parsed_data_pointer).model.as_ptr()).to_str().unwrap();
```

Comment:
Would it be possible to remove the unwraps here? It'll blow up the caller if any of them fail and I try to avoid that in libraries.

Reply:
I changed the unwraps to set the strings to "" in the struct. Since it's already returning an error based on the C function I feel as though I can't return anything related to a problem of casting the C strings. Is that ok, or should there be some why of telling the function caller there was an issue?

Reply:
I was thinking something like:
```
CStr::from_ptr((*parsed_data_pointer).model.as_ptr()).to_str().map_err(|_| Errno::EINVAL)?; 
```
It swallows the utf8 parsing error but it does tell the caller that something went wrong. Maybe we could add a document comment to say if you get `EINVAL` then some string parsing error happened. It's a bit unfortunate that `Errno` doesn't really allow that much to be said about what happened.

Reply:
ah, okay, I like having it still return something. I've been looking into `Errno` codes, and you're right, there isn't much we can use. What about using `ENOTRECOVERABLE`? Just because `EINVAL` could be confused for something user related. Since it means "invalid argument" and maybe a user could think they gave it something invalid as an argument. But then again, counter arguing from what you said, that could be solved by adding a comment. hmmmmm, I feel as though it wont matter in the end, I'll ust `EINVAL`